### PR TITLE
Add missing headers in Visual Studio

### PR DESCRIPTION
**Pull request by SeoHyungjun**
dirent.h and dlfcn.h are headers not provided by visual studio 2022.

Added for use in nntrainer.

### CODE REVIEW

File: external/dlfcn-win32/dlfcn.h [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+DLFCN_EXPORT char *dlerror(void);
+
+/* Translate address to symbolic information (no POSIX standard) */
+DLFCN_EXPORT int dladdr(const void *addr, Dl_info *info);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* DLFCN_H */
```
Comment (baek2sm) [inline external/dlfcn-win32/dlfcn.h]: It seems to be needed new line at the end.

File: external/dirent/dirent.h [hunk 2]
```diff
@@ -0,0 +1,1212 @@
...
+{
+	/* Non-Microsoft compiler or older Microsoft compiler */
+	errno = error;
+}
+#endif
+
+#ifdef __cplusplus
+}
+#endif
+#endif /*DIRENT_H*/
```
Comment (baek2sm) [inline external/dirent/dirent.h]: It seems to be needed new line at the end.

**Review by baek2sm — APPROVED** [review]
LGTM

**Review by DonghakPark — APPROVED** [review]
LGTM!

File: external/dlfcn-win32/dlfcn.h (hunk 1 above)
Comment (SeoHyungjun) [reply external/dlfcn-win32/dlfcn.h]: Thank you!

**Pull request merged**